File: Bronze1-1/06-22-am/unit05/word-lengths/length.java
/*
 * Word length product
 */
import java.util.*;
public class length {
    public static void main(String[] args) {
    	Scanner in = new Scanner(System.in);
    	int N = in.nextInt(); //# of words
    	String[] words = new String[N];
    	for(int j=0; j<N; j++)
    		words[j] = in.next();
    	in.close();
    	
    	int ans = 0;

    	//freq[j]: frequency of the letters in words[j]
    	int[][] freq = new int[N][26];
        for(int i=0; i<N; i++) {
            char[] A = words[i].toCharArray();
            for(int j=0; j<A.length; j++) {
                int c = A[j] - 'a';
                freq[i][c]++;
            }
        }

        for(int i=0; i<words.length-1; i++) {
            LABEL:
            for(int j=i+1; j<words.length; j++) {
                for(int k = 0; k < 26; k++) {
                    if(freq[i][k] != 0 && freq[j][k] != 0)
                        continue LABEL;
                }
                ans = Math.max(ans, words[i].length() * words[j].length());
            }
        }
        
        System.out.println(ans);
    }
}

File: Bronze1-1/06-22-am/unit05/word-lengths/length.cpp
/*
 * Word length product
 */
#include <bits/stdc++.h>
using namespace std;

int main() {
	int N; cin >> N; //# of words
	string words[N];
	for(int j=0; j<N; j++)
		cin >> words[j];
	
	int ans = 0;

	//freq[j]: frequency of the letters in words[j]
	int freq[N][26];
	for(int i=0; i<N; i++)
		fill(freq[i], freq[i]+26, 0);
		
    for(int i=0; i<N; i++) {
        for(int j=0; j<words[i].length(); j++) {
            int c = words[i][j] - 'a';
            freq[i][c]++;
        }
    }

    for(int i=0; i<N-1; i++) {
        for(int j=i+1; j<N; j++) {
        	bool overlap = false;
            for(int k=0; k<26; k++) {
                if(freq[i][k] != 0 && freq[j][k] != 0) {
					overlap = true;
					break;
				}
            }
            if(!overlap)
	            ans = max(ans, (int)words[i].length() * (int)words[j].length());
        }
    }
    
    cout << ans << '\n';
}

